### PR TITLE
Feature/schedule !! Sorry for the delay!

### LINKED ISSUE #5: Schedule
**Is your feature request related to a problem? Please describe.**
As a student, I want to see the schedule of my class but also other events of the school (like when is the last day to cancel a course)

**Describe the solution you'd like**
A screen I can access from easily in the application

### PR DESCRIPTION
**Pull request by apomalyn**
# Finally!! 
I'm really sorry for the size of the PR, this is not a good practice at all...

## Checklist
- [x] Every manager and services are tested
- [x] Every manager and services are documented
- [x] Every widget is tested
- [x] Every view is tested
- [x] Every viewmodel is tested 

## Issues related
Closes #5 

## Results
https://user-images.githubusercontent.com/22211097/104272683-ad42c480-546b-11eb-97aa-d4411cfa4661.mov

P.S.: the commit author change halfway but it's still me, I cloned again the project and forgot to correctly set my author name. And because some of my commits have been already used on other branches, amend the entire history will cause some damage...

### CODE REVIEW

**Review by AlexandreMarkus — COMMENTED** [review]
review in progress

**Review by AlexandreMarkus — COMMENTED** [review]
review in progress

**Review by AlexandreMarkus — CHANGES_REQUESTED** [review]
Courses should be listed by time instead of by name

image

File: lib/core/viewmodels/schedule_settings_viewmodel.dart [hunk 1]
```diff
@@ -0,0 +1,78 @@
+import 'package:enum_to_string/enum_to_string.dart';
```
Comment (AlexandreMarkus) [inline lib/core/viewmodels/schedule_settings_viewmodel.dart:1]: optimize imports

File: lib/locator.dart [hunk 2]
```diff
@@ -2,15 +2,20 @@
 import 'package:flutter_secure_storage/flutter_secure_storage.dart';
 import 'package:get_it/get_it.dart';
 import 'package:http/http.dart' as http;
+import 'package:logger/logger.dart';
+import 'package:notredame/core/managers/settings_manager.dart';
```
Comment (AlexandreMarkus) [inline lib/locator.dart:6]: place under good section @apomalyn

File: lib/ui/views/schedule_view.dart [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+            ),
+          ),
+        ),
+      ),
+    );
+  }
+
+  /// Build the calendar
+  Widget _buildTableCalendar(ScheduleViewModel model) => TableCalendar(
+        //TODO uncomment when https://github.com/aleksanderwozniak/table_calendar/issues/164 is close
```
Comment (AlexandreMarkus) [inline lib/ui/views/schedule_view.dart:135]: reminder @apomalyn

Comment (apomalyn) [reply lib/ui/views/schedule_view.dart:135]: That is a reminder, this TODO should stay there until the PR in tablecalendar is closed

File: lib/ui/widgets/schedule_settings.dart [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+        title: Text(getTextForFormat(context, format)),
+      ));
+    }
+
+    tiles.add(const Divider(thickness: 1));
+
+    return tiles;
+  }
+
+  // TODO remove ignore when https://github.com/aleksanderwozniak/table_calendar/issues/164 is close
```
Comment (AlexandreMarkus) [inline lib/ui/widgets/schedule_settings.dart:116]: reminder @apomalyn

**Comment by AlexandreMarkus** [comment]
Which: Golden "goldenFiles/scheduleView_1.png": Pixel test failed, 0.48% diff detected.

All golden tests are failing.

image

Got the same results on my PC

File: lib/core/viewmodels/schedule_settings_viewmodel.dart (hunk 1 above)
Comment (apomalyn) [reply lib/core/viewmodels/schedule_settings_viewmodel.dart:1]: Nice catch 👍

File: lib/ui/widgets/schedule_settings.dart (hunk 4 above)
Comment (apomalyn) [reply lib/ui/widgets/schedule_settings.dart:116]: Same as the precedent.  This TODO should stay there until the PR in tablecalendar is closed but nice catch! Nothing pass your eyes ahha :D

**Comment by apomalyn** [comment]
> Which: Golden "goldenFiles/scheduleView_1.png": Pixel test failed, 0.48% diff detected.
> 
> All golden tests are failing.
> 
> image
> 
> Got the same results on my PC

Fixed. Thanks!

**Comment by apomalyn** [comment]
> Courses should be listed by time instead of by name
> 
> image

Fixed!

Screen Shot 2021-01-16 at 1 14 55 PM

**Comment by AlexandreMarkus** [comment]
> > Which: Golden "goldenFiles/scheduleView_1.png": Pixel test failed, 0.48% diff detected.
> > All golden tests are failing.
> > image
> > Got the same results on my PC
> 
> Fixed. Thanks!

I did restart the checks and goldenFiles are failing the tests

**Comment by apomalyn** [comment]
> > > Which: Golden "goldenFiles/scheduleView_1.png": Pixel test failed, 0.48% diff detected.
> > > All golden tests are failing.
> > > image
> > > Got the same results on my PC
> > 
> > 
> > Fixed. Thanks!
> 
> I did restart the checks and goldenFiles are failing the tests

This is really weird! Yesterday they were passing! I really have to get the bottom of this.

**Review by AlexandreMarkus — APPROVED** [review]

**Pull request merged**